### PR TITLE
[ng] Dropdown focus backport

### PR DESCRIPTION
**Pull request by jeeyun**
Backporting #3400, #3450, #3488, #3501 to 1.x.

Note that there's a very old, undocumented version of dropdown markup that hasn't been supported since before v0.10 (i.e. using `<div class="dropdown-menu"></div>` in the Angular component). This backport will NOT provide keyboard navigation for that deprecated markup.

### CODE REVIEW

**Comment by gnomeontherun** [comment]
Just to be clear, while it doesn't support the old markup (understandable), does it break its behavior?

**Comment by jeeyun** [comment]
@gnomeontherun - it doesn't break its previous behavior, but preserves it. Before this PR, the old markup did not support keyboard navigation (does not respond to arrow keys, etc). After this PR, the behavior will remain the same for the old markup.

File: src/clr-angular/popover/dropdown/dropdown-trigger.ts [hunk 1]
```diff
@@ -17,16 +18,24 @@ import { ClrDropdown } from './dropdown';
     '[class.dropdown-item]': '!isRootLevelToggle',
     '[class.expandable]': '!isRootLevelToggle',
     '[class.active]': 'active',
+    '[attr.aria-haspopup]': '"menu"',
+    '[attr.aria-expanded]': 'ifOpenService.open',
```
Comment (gnomeontherun) [inline src/clr-angular/popover/dropdown/dropdown-trigger.ts]: We need to apply the same fix as #3450.

Comment (jeeyun) [reply src/clr-angular/popover/dropdown/dropdown-trigger.ts]: Absolutely. I'm also merging fixes from https://github.com/vmware/clarity/pull/3440 into this PR.

**Pull request merged**